pkg/client/dtclient/dummy_client.go

```diff
@@ -246,8 +247,17 @@ func (c *DummyClient) ConfigExistsByName(a api.API, name string) (exists bool, i
 }
 
 func (c *DummyClient) UpsertSettings(obj SettingsObject) (DynatraceEntity, error) {
+
+	id := obj.Id
+
+	// to ensure decoding of Management Zone Numeric IDs works for dry-runs the dummy client needs to produce a fake but validly formated objectID
+	if obj.SchemaId == "builtin:management-zones" {
```

Comment:
note: Theoretically I think we could always do this and save the `if`, but I wasn't sure which option is better - be explicit about the MZ handling or keeping the dummy client simpler